lib/generators/declare_schema/migration/migration_generator.rb

```diff
@@ -117,6 +106,36 @@ def migrations_pending?
       end
     end
 
+    def load_migrations
+      if ActiveSupport.version >= Gem::Version.new('7.1.0')
+        ActiveRecord::MigrationContext.new(
+          ActiveRecord::Migrator.migrations_paths,
+          ActiveRecord::Base.connection.schema_migration,
+          ActiveRecord::Base.connection.internal_metadata
+        ).migrations
+      elsif ActiveSupport.version >= Gem::Version.new('6.0.0')
+        ActiveRecord::MigrationContext.new(ActiveRecord::Migrator.migrations_paths, ActiveRecord::SchemaMigration).migrations
+      else # Rails 5.2 and earlier
```

Comment:
❓ Should we go head and rip out Rails 5 support from the gem and make this a major version bump instead?

Reply:
Yeah, I'd say so. I think in general its common practice to only support 2 major versions at a time. This seems reasonable to me that we'd also only support two major versions.

Reply:
Done 189372263b0e6a2fae3a6b51fa5e992507a7012c